src/node.js

```diff
@@ -116,10 +116,27 @@
 
       // If -i or --interactive were passed, or stdin is a TTY.
       if (process._forceRepl || NativeModule.require('tty').isatty(0)) {
+        var history;
+        var path = NativeModule.require('path');
+        var fs = NativeModule.require('fs');
+        var HISTORY_PATH = path.join(process.env.HOME || process.env.USERPROFILE,
```

Comment:
I think this should live in `os` module as `os.homedir()`. It requires a separate PR and some discussion though (and TC involvement)